src/d3d11/d3d11_blend.cpp

```diff
@@ -148,8 +148,130 @@ namespace dxvk {
   
   
   HRESULT D3D11BlendState::NormalizeDesc(D3D11_BLEND_DESC1* pDesc) {
-    // TODO validate
-    // TODO clear unused values
+    const D3D11_BLEND_DESC1 defaultDesc = DefaultDesc();
+    if (pDesc->AlphaToCoverageEnable != 0) {
+      pDesc->AlphaToCoverageEnable = 1;
+    }
+
+    if (pDesc->IndependentBlendEnable != 0) {
+      pDesc->IndependentBlendEnable = 1;
+    }
+
+    D3D11_RENDER_TARGET_BLEND_DESC1* rt = &pDesc->RenderTarget[0];
+    if (rt->BlendEnable != 0) {
+      rt->BlendEnable = 1;
+      //Can not have Blend enabled the same time as LogicOp
+      if (rt->LogicOpEnable == 1) {
+        Logger::err(str::format("D3D11BlendState: Logic Op must be disabled if Blend is enabled: "));
+        return E_INVALIDARG;
+      }
+      
+      if (!ValidateBlendOperations(
+          rt->SrcBlend, rt->SrcBlendAlpha,
+          rt->DestBlend, rt->DestBlendAlpha,
+          rt->BlendOp, rt->BlendOpAlpha))
+        return E_INVALIDARG;
+    }
+
+    if (rt->LogicOpEnable != 0) {
+      rt->LogicOpEnable = 1;
+
+      if (rt->BlendEnable == 1) {
+        Logger::err(str::format("D3D11BlendState: Blending must be disabled if LogicOp is enabled: "));
+        return E_INVALIDARG;
+      }
+       
+      if (pDesc->IndependentBlendEnable == 1) {
+        Logger::err(str::format("D3D11BlendState: IndependentBlendEnable must be disabled if LogicOp is enabled: "));
+        return E_INVALIDARG;
+      }
+      
+      if (!ValidLogicOp(rt->LogicOp)) {
+        Logger::err(str::format("D3D11BlendState: Invalid LogicOp: ", rt->LogicOp));
+        return E_INVALIDARG;
+      }
+        
+    }
+    if (rt->BlendEnable == 0) {
+      rt->SrcBlend = defaultDesc.RenderTarget[0].SrcBlend;
+      rt->DestBlend = defaultDesc.RenderTarget[0].DestBlend;
+      rt->BlendOp = defaultDesc.RenderTarget[0].BlendOp;
+      rt->SrcBlendAlpha = defaultDesc.RenderTarget[0].SrcBlendAlpha;
+      rt->DestBlendAlpha = defaultDesc.RenderTarget[0].DestBlendAlpha;
+      rt->BlendOpAlpha = defaultDesc.RenderTarget[0].BlendOpAlpha;
+    }
+
+    if (rt->LogicOpEnable == 0) {
+      rt->LogicOp = defaultDesc.RenderTarget[0].LogicOp;
+    }
+
+    if (rt->RenderTargetWriteMask > D3D11_COLOR_WRITE_ENABLE_ALL) {
+      Logger::err(str::format("D3D11BlendState: Invalid RenderTargetWriteMask: ", rt->RenderTargetWriteMask));
+      return E_INVALIDARG;
+    }
+
+    //for the rest of the rendertargets
```

Comment:
This loop looks like unnecessary code duplication, or what are the differences between the first and all following render targets?